### ISSUE TITLE
iOS crash - Delegate must respond to locationManager:didFailWithError:

### ISSUE THREAD

**Issue by chris-wickens** [mapbox/mapbox-maps-flutter#413]
On version 1.0.0-beta.3 my iOS app is crashing in the simulator after a few moments when the location is set to freeway drive.

Steps to reproduce:

1. Launch the example app in the iOS simulator (tested on iPhone 15 Pro iOS 17 & 17.2, Xcode 15.2)
2. In the simulator's Features menu set the location to freeway drive
3. Open the Location Component example screen
4. Click "show location"
5. After a few seconds the app should crash

I can't confirm that this happens on physical devices but the error message makes me think that it's a possibility.

Stacktrace:
```
*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Delegate must respond to locationManager:didFailWithError:'
*** First throw call stack:
(
	0   CoreFoundation                      0x0000000180491128 __exceptionPreprocess + 172
	1   libobjc.A.dylib                     0x000000018008412c objc_exception_throw + 56
	2   Foundation                          0x0000000180d1163c _userInfoForFileAndLine + 0
	3   CoreLocation                        0x000000018406621c CLClientStopVehicleHeadingUpdates + 60756
	4   MapboxCommon                        0x0000000101f6b594 _ZN6mapbox7bindgen6mapbox6common8location23LocationServiceObserver30onAccuracyAuthorizationChangedENS_6common8location21AccuracyAuthorizationE + 5512
	5   CoreLocation                        0x0000000184075160 CLClientStopVehicleHeadingUpdates + 122008
	6   CoreLocation                        0x0000000184071f98 CLClientStopVehicleHeadingUpdates + 109264
	7   CoreLocation                        0x0000000184050114 CLClientCreateWithBundleIdentifierAndPathWithWebsiteOnSilo + 5956
	8   LocationSupport                     0x000000018a9eb47c _ZN18CLConnectionServer13handleMessageEP12CLConnectionNSt3__110shared_ptrI19CLConnectionMessageEE + 17436
	9   CoreFoundation                      0x00000001803f1ce0 __CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__ + 20
	10  CoreFoundation                      0x00000001803f1428 __CFRunLoopDoBlocks + 352
	11  CoreFoundation                      0x00000001803ebca8 __CFRunLoopRun + 752
	12  CoreFoundation                      0x00000001803eb5a4 CFRunLoopRunSpecific + 572
	13  CoreFoundation                      0x00000001803ec46c CFRunLoopRun + 60
	14  MapboxCommon                        0x0000000101f6e774 _ZN6mapbox7bindgen6mapbox6common8location23LocationServiceObserver30onAccuracyAuthorizationChangedENS_6common8location21AccuracyAuthorizationE + 18280
	15  Foundation                          0x0000000180df3e70 __NSThread__start__ + 720
	16  libsystem_pthread.dylib             0x000000010133b4c0 _pthread_start + 104
	17  libsystem_pthread.dylib             0x00000001013366f0 thread_start + 8
)
libc++abi: terminating due to uncaught exception of type NSException
```

**Comment by baleboy** [comment]
Thanks for the report, we are aware of this issue and it will be resolved in 1.0.0-rc.1 due today.

**Comment by baleboy** [comment]
Fixed in https://github.com/mapbox/mapbox-maps-flutter/releases/tag/1.0.0-rc.1
